source/common/router/router.cc

```diff
@@ -90,6 +90,21 @@ bool convertRequestHeadersForInternalRedirect(Http::HeaderMap& downstream_header
   return true;
 }
 
+constexpr uint64_t TimeoutPrecisionFactor = 100;
+
+// Express percentage as [0, TimeoutPrecisionFactor] because stats do not accept floating point
+// values, and getting multiple significant figures on the histogram would be nice.
+uint64_t percentageOfTimeout(const std::chrono::milliseconds response_time,
+                             const std::chrono::milliseconds timeout) {
+  // Timeouts of 0 are considered infinite. Any portion of an infinite timeout used is still
+  // none of it.
+  if (timeout.count() == 0) {
+    return 0;
+  }
+
+  return static_cast<uint64_t>(response_time.count() * TimeoutPrecisionFactor / timeout.count());
```

Comment:
nit: I think this static_cast is unnecessary since the function already returns a uint64_t.

Reply:
It's there for annotation that there is intentional narrowing in the absence of a `gsl::narrow_cast` type call.